service/s3.h

```diff
@@ -261,9 +259,21 @@ struct S3Api : public AwsApi {
         std::string contentEncoding;
         std::map<std::string, std::string> metadata;
         std::string acl;
+    };
+
+    /** A set of options that are used when performing uploads. */
+    struct UploadOptions {
```

Comment:
Because canned acls are just a shortcut for setting up real acls. They do not really exist. When yhou specify them: s3 translates them into real acls on the fly.